src/UpdatedPackagesCollector.js

```diff
@@ -127,7 +127,19 @@ export default class UpdatedPackagesCollector {
 
   collectUpdates() {
     return this.packages.filter(pkg => {
-      return this.updatedPackages[pkg.name] || (this.flags.onlyExplicitUpdates ? false : this.dependents[pkg.name]) || this.flags.canary;
+      if (this.updatedPackages[pkg.name]) {
+        return true;
+      }
+
+      if (!this.flags.onlyExplicitUpdates && this.dependents[pkg.name]) {
+        return true;
+      }
+
+      if (this.flags.canary && !pkg.isPrivate()) {
+        return true;
```

Comment:
No I still want canary to work even if some modules are private. My idea is to not implement the check in this method at all. Instead we should implement it when we: prompt new versions (no need to bump a new semver for private modules IMO) and when we run git/npm commands. This way I think we can get the dependency updates to work but skip git tags, version bumping (dependency bumping will still work) and npm publish.

e.g here: https://github.com/lerna/lerna/blob/master/src/commands/PublishCommand.js#L385
https://github.com/lerna/lerna/blob/master/src/commands/PublishCommand.js#L336
https://github.com/lerna/lerna/blob/master/src/commands/PublishCommand.js#L227
https://github.com/lerna/lerna/blob/master/src/commands/PublishCommand.js#L173
...and probably some other places.. 

Should probably change `this.updates` to something else for these, e.g `this.publicUpdates`